### ISSUE TITLE
Cooperative scroll mode

### ISSUE THREAD

**Issue by cmahnke** [openseadragon/openseadragon#2597]
This is a feature request and follow up of discussion item #2596:
Google Maps has a so called [cooperative mode](https://developers.google.com/maps/documentation/javascript/interaction#gesturehandling_cooperative) which is really helpful on mobile devices: It disables the one touch drag in favour of keeping the (drag) navigation of the surrounding page working.
If you try to use one finger you get a message, that you need to use two fingers. The link also provides an example.

Other sites / tools also support such a mode:
* [MapTiler](https://docs.maptiler.com/sdk-js/examples/cooperative-gestures/)
* Leaflet Plugin [GestureHandling](https://github.com/elmarquis/Leaflet.GestureHandling)
* OpenLayers [example](https://openlayers.org/en/latest/examples/two-finger-pan-scroll.html?q=scroll)
* A quite old Stack Overflow [discussion](https://stackoverflow.com/questions/42769202/openlayer-two-finger-movement-dragpan) regarding OpenLayers

**Comment by Aiosa** [comment]
This could be useful for webpages, but not so nice if the viewport itself si the (whole) page. So I can imagine this as an optional feature.

**Comment by cmahnke** [comment]
Yes of course, it was not my intention to have it always active.

**Comment by iangilman** [comment]
@cmahnke Thank you for filing this! If you're interested in working on it, I can help point you in the right direction. Otherwise hopefully someone else will take an interest in it.

**Comment by msalsbery** [comment]
It's a relatively easy enhancement to the supported gestures. I'll take it if nobody else is on it

**Comment by cmahnke** [comment]
@msalsbery That would be great!

**Comment by pearcetm** [comment]
@msalsbery I'm building a site where a non-trivial amount of traffic is likely to be on mobile, and I'd love to include such a feature for when a mobile platform is detected so users can more easily scroll past the OpenSeadragon viewer. Looking forward to this enhancement!

**Comment by iangilman** [comment]
@msalsbery any news on this? If you haven't gotten to it, should someone else take it on?

**Comment by msalsbery** [comment]
@cmahnke @pearcetm @iangilman 

How far would we want to take this in OpenSeadragon? The pointer event gesture handling is relatively straightforward, but the UI is something we've always left to the user.

Maybe add some events so the user can display their own UI? What events would be useful besides the obvious one-finger touch when it's in cooperative mode?

Add your wish list items now 😉

**Comment by cmahnke** [comment]
@msalsbery 
From my point of view it's depending on use case priorities: Is OpenSeadragon expected to be a component that's expected to be exposed directly to a end user or is it always considered to be a library used by other viewers which in turn are used by end users. Or in other words: Is it some sort of widget or a library to build your own widgets. Certainly both use cases a valid, but if I  it's considered to be more like a UI library I would agree: Leave it to the user of the library (which is not the end user).

Before trying to make up some use cases, for UI gestures, I would just have a look at similar interface (mainly for maps) extract gesture features there, if they make sense (so skip things like pitch).

**Comment by iangilman** [comment]
This is a common enough request that I think it is worth having in the OSD library. I'm imagining once you turn on the flag for it, the only UI is the message you get if you drag with one finger. I think we should provide a default version of that, but also provide a hook for the app developer to put up their own UI.

@msalsbery Great that you're tackling this!

**Comment by msalsbery** [comment]
@iangilman 

Cool, that's the direction I was leaning, thanks!

@cmahnke 

Thank you for the quick reply! I was just referring to the actual presentation of UI elements, something we're always hesitant to do. I rarely see real world usage that uses our default UI 😅 

I'm definitely taking a look at other examples...if you have more links I will certainly appreciate and look at them!

**Comment by iangilman** [comment]
Fixed by #2934
